core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -74,6 +78,23 @@ oppia.factory('responsesService', [
           editorContextService.getActiveStateName(),
           angular.copy(newAnswerGroups));
 
+        if (
+          stateInteractionIdService.savedMemento &&
+          INTERACTION_SPECS[stateInteractionIdService.savedMemento]
+            .can_have_solution && stateSolutionService.savedMemento !== null) {
+          if (stateSolutionService.savedMemento.correctAnswer !== null) {
+            var currentStateName = editorContextService.getActiveStateName();
+            SolutionHelperService.verifySolution(
+              explorationContextService.getExplorationId(),
+              explorationStatesService.getState(currentStateName),
+              stateSolutionService.savedMemento.correctAnswer,
+              function () {
+                alertsService.addInfoMessage(
+                  'That solution does not lead to the next state!');
```

Comment:
Depends on whether we have a flag feature -- flag isn't working as of now. If there is no flag we could remind the creator that the solution isn't correct (with the current rules) on each update.